src/settings.js

```diff
@@ -166,31 +166,56 @@ const settings = {
         await match.accessible.ButtonBarButton('Done').tap();
     },
     /**
-     * @name settings.CorrectionRange
-     * @param {corrections} corrections e.g. { range: { min: '80', max: '150' } });
-     * corrections:{
-     *    range:{ min: string, max: string,},
-     *     preMeal:{min: string, max: string},
-     *     workout:{min: string, max: string}
-     * },
+     * @name settings.CorrectionRanges
+     * @param {object} ranges e.g. [{ time: '12:00 AM', min: '80', max: '150' }];
      */
-    async CorrectionRange(corrections) {
-        console.log('corrections: ', corrections);
+    async CorrectionRanges(ranges) {
+        const correctionRangePickerColumns = {
+            Units: 5,
+            MaximumValue: 4,
+            Separator: 3,
+            MinimumValue: 2,
+            Time: 1,
+        };
+
         await match.accessible.ButtonBarButton('Settings').tap();
         await match.accessible.Text('Correction Range').tap();
         await match.accessible.ButtonBarButton('Add').tap();
 
-        if (corrections.range) {
-            console.log('corrections.range: ', corrections.range);
-            //TODO: atIndex!! need a better way
-            // await match.accessible.Label('12:00 AM').atIndex(0).tap();
-            // await match.accessible.Label(`${corrections.range.max}`).atIndex(1).tap();
-            // await match.accessible.Label(`${corrections.range.min}`).atIndex(0).tap();
-            // await match.accessible.Label('Save').tap();
+        if (ranges) {
+            let pickerItemIndex = 1;
+            for (let index = 0; index < ranges.length; index++) {
+                const range = ranges[index];
+                await match.accessible.Label(`${range.time}`).atIndex(0).tap();
```

Comment:
This assumes that the ranges are time ordered correctly, yes? If so, might want to specify that in above documentation so it is clear or check it in code.